src/utils/simulation_results.jl

```diff
@@ -0,0 +1,170 @@
+variable = [:P_ThermalStandard, :P_RenewableDispatch]
+results = load_simulation_results(stage, step, variable, SimulationResultsReference)
+```
+# Accepted Key Words
+- `write::Bool`: if true, the aggregated results get written back to the results file in the folder structure
+"""
+
+function load_simulation_results(SimulationResultsReference::SimulationResultsReference, stage_number::Int; kwargs...)
+    stage = "stage-$stage_number"
+    references = SimulationResultsReference.ref
+    variables = Dict()
+    duals = Dict()
+    variable = (collect(keys(references[stage])))
+    dual = _find_duals(variable)
+    variable = setdiff(variable, duals)
+    time_stamp = DataFrames.DataFrame(Range = Dates.DateTime[])
+    time_length = SimulationResultsReference.chronologies[stage]
+
+    for l in 1:length(variable)
+        date_df = references[stage][variable[l]]
+        variables[(variable[l])] = DataFrames.DataFrame()
+        for (ix,time) in enumerate(date_df.Date)
+            file_path = date_df[ix, :File_Path]
+            var = Feather.read(file_path)
+            variables[(variable[l])] = vcat(variables[(variable[l])],var[1:time_length,:])
+            if l == 1
+                time_stamp = vcat(time_stamp, _reading_time(file_path, time_length))
+                check_file_integrity(dirname(file_path))
+            end
+        end
+    end
+    time_stamp[!,:Range] = convert(Array{Dates.DateTime}, time_stamp[!,:Range])
+    file_path = dirname(references[stage][variable[1]][1,:File_Path])
+    optimizer = JSON.parse(open(joinpath(file_path, "optimizer_log.json")))
+    obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer["obj_value"])
+    if !isempty(dual)
+        duals = _reading_references(duals, dual, stage, references, extra_time_length)
+        results = DualResults(variables, obj_value, optimizer, time_stamp, duals)
+    else
+        results = SimulationResults(variables, obj_value, optimizer, time_stamp)
+    end
+    file_type = get(kwargs, :file_type, Feather)
+    write = get(kwargs, :write, false)
+    if write == true
+        write_results(results, SimulationResultsReference.results_folder, "results")
+    end
+    return results
+end
+"""
+    check_file_integrity(path::String)
+
+Checks the hash value for each file made with the file is written with the new hash_value to verify the file hasn't been tampered with since written
+
+# Arguments
+- `path::String`: this is the folder path that contains the results and the check.sha256 file
+"""
+function check_file_integrity(path::String)
+    file_path = joinpath(path, "check.sha256")
+    text = open(file_path, "r") do io
+        return readlines(io)
+    end
+    for line in text
+        hash_value, file_name = split(line)
+        if String(hash_value) !== String(bytes2hex(SHA.sha256(open(file_name))))
+            throw(IS.DataFormatError("The hash value in the written files does not match the read files, results may have been tampered."))
```

Comment:
Sorry for not noticing this earlier, but I have two comments:

1. Either the exception message or a log [error] message should indicate the bad filename, expected hash, and actual hash.
2. It might be good to check all files so that the log file shows all the ones that are bad.  You could throw the exception after concluding the loop.  That's just my opinion, I'll leave it to you.